nixos/modules/services/web-apps/funkwhale/requirements.nix

```diff
@@ -0,0 +1,1788 @@
+      doCheck = commonDoCheck;
+      buildInputs = commonBuildInputs;
+      propagatedBuildInputs = [ ];
+      meta = with pkgs.stdenv.lib; {
+        homepage = "http://github.com/andymccurdy/redis-py";
+        license = licenses.mit;
+        description = "Python client for Redis key-value store";
+      };
+    };
+
+
+
+    "requests" = python.mkDerivation {
+      name = "requests-2.21.0";
+      src = pkgs.fetchurl { url = "https://files.pythonhosted.org/packages/52/2c/514e4ac25da2b08ca5a464c50463682126385c4272c18193876e91f4bc38/requests-2.21.0.tar.gz"; sha256 = "502a824f31acdacb3a35b6690b5fbf0bc41d63a24a45c4004352b0242707598e"; };
+      doCheck = commonDoCheck;
+      buildInputs = commonBuildInputs;
+      propagatedBuildInputs = [
+      self."certifi"
+      self."chardet"
+      self."cryptography"
+      self."idna"
+      self."urllib3"
+    ];
+      meta = with pkgs.stdenv.lib; {
+        homepage = "http://python-requests.org";
+        license = licenses.asl20;
+        description = "Python HTTP for Humans.";
+      };
+    };
+
+
+
+    "requests-http-signature" = python.mkDerivation {
+      name = "requests-http-signature-0.1.0";
+      src = pkgs.fetchurl { url = "https://files.pythonhosted.org/packages/10/18/3744e7bc8b846dc7f4718ad189261f2eb47a70190a2f69a877f2ab51a77c/requests-http-signature-0.1.0.tar.gz"; sha256 = "0e39d928469e6f1411e3bffca74a280ac9375d4fa5bf03552974f0ba4ff4c37a"; };
+      doCheck = commonDoCheck;
+      buildInputs = commonBuildInputs;
+      propagatedBuildInputs = [
+      self."cryptography"
+      self."requests"
+    ];
+      meta = with pkgs.stdenv.lib; {
+        homepage = "https://github.com/kislyuk/requests-http-signature";
+        license = "License :: OSI Approved :: Apache Software License";
+        description = "A Requests auth module for HTTP Signature";
+      };
+    };
+
+
+
+    "requests-oauthlib" = python.mkDerivation {
+      name = "requests-oauthlib-1.0.0";
+      src = pkgs.fetchurl { url = "https://files.pythonhosted.org/packages/95/be/072464f05b70e4142cb37151e215a2037b08b1400f8a56f2538b76ca6205/requests-oauthlib-1.0.0.tar.gz"; sha256 = "8886bfec5ad7afb391ed5443b1f697c6f4ae98d0e5620839d8b4499c032ada3f"; };
+      doCheck = commonDoCheck;
+      buildInputs = commonBuildInputs;
+      propagatedBuildInputs = [
+      self."oauthlib"
+      self."requests"
+    ];
+      meta = with pkgs.stdenv.lib; {
+        homepage = "https://github.com/requests/requests-oauthlib";
+        license = licenses.bsdOriginal;
+        description = "OAuthlib authentication support for Requests.";
+      };
+    };
+
+
+
+    "rsa" = python.mkDerivation {
+      name = "rsa-4.0";
+      src = pkgs.fetchurl { url = "https://files.pythonhosted.org/packages/cb/d0/8f99b91432a60ca4b1cd478fd0bdf28c1901c58e3a9f14f4ba3dba86b57f/rsa-4.0.tar.gz"; sha256 = "1a83640640573
```

Comment:
Read https://nixos.org/nixpkgs/manual/#contributing on how to contribute python packages. pypi2nix isn't desired because of these large files. Take a look at other python packages.